en/advanced-topics/jsb-auto-binding.md

```diff
@@ -0,0 +1,374 @@
+    return false;
+}
+SE_BIND_FUNC(js_cocos2dx_test_JSBBridge_getInstance)
+
+bool js_register_cocos2dx_test_JSBBridge(se::Object* obj) // NOLINT(readability-identifier-naming)
+{
+    auto* cls = se::Class::create("JSBBridge", obj, nullptr, nullptr);
+
+    cls->defineFunction("abcLog", _SE(js_cocos2dx_test_JSBBridge_abcLog));
+    cls->defineStaticFunction("getInstance", _SE(js_cocos2dx_test_JSBBridge_getInstance));
+    cls->install();
+    JSBClassType::registerClass<abc::JSBBridge>(cls);
+
+    __jsb_abc_JSBBridge_proto = cls->getProto();
+    __jsb_abc_JSBBridge_class = cls;
+
+    se::ScriptEngine::getInstance()->clearException();
+    return true;
+}
+bool register_all_cocos2dx_test(se::Object* obj)
+{
+    // Get the ns
+    se::Value nsVal;
+    if (!obj->getProperty("abc", &nsVal))
+    {
+        se::HandleObject jsobj(se::Object::createPlainObject());
+        nsVal.setObject(jsobj);
+        obj->setProperty("abc", nsVal);
+    }
+    se::Object* ns = nsVal.toObject();
+
+    js_register_cocos2dx_test_JSBBridge(ns);
+    return true;
+}
+```
+
+Doesn't it look familiar? It's exactly the same as Cocos' existing `.cpp` files, even including the registration functions and class definitions that are all automatically generated.
+
+## Cocos Compilation Configuration
+
+Although we have generated the bindings after the above step, the Typescript layer can't be used directly because we still need to configure the generated bindings into the `CMakeLists.txt` file to be compiled with other C++ files, which is the last part of the `CMakeLists.txt` compilation configuration.
+
+1. Open the `CMakeLists.txt` file and add the initial `ABCJSBBridge.h` and `ABCJSBBridge.cpp` to it, as well as the `jsb_cocos2dx_test_auto.h` and `jsb_cocos2dx_test_auto.cpp` files generated by the automatic bindings:
+
+    ![](jsb/111.png)
+
+2. Open `cocos/bindings/manual/jsb_module_register.cpp` and add the registration code for the cocos2dx_test module:
+
+    ![](jsb/112.png)
+
+
+After the above configuration, call it directly from the ts layer like this:
+
+``` typescript
+import { _decorator, Component, Node } from 'cc';
+const { ccclass, property } = _decorator;
+
+@ccclass('Test')
+export class Test extends Component {
+    start () {
+        // @ts-ignore
+        abc.JSBBridge.getInstance().abcLog("JSB binding test success")
+    }
+}
+```
+
+## Restrictions on Auto-Binding
+
+Auto-binding relies on the [Bindings Generator tool](https://github.com/cocos-creator/bindings-generator). The Bindings Generator tool can Bind public methods and public properties of C++ classes to the scripting layer. The automatic binding tool, although very powerful, has a few limitations:
+1. it can only generate bindings for classes, not structs, independent functions, etc.
+2. it is not possible to generate `Delegate` type APIs, because objects in scripts cannot inherit from the `Delegate` class in C++ and override the `Delegate` functions in it.
```

Comment:
```suggestion
2. It is not possible to generate `Delegate` type APIs, because objects in scripts cannot inherit from the `Delegate` class in C++ and override the `Delegate` functions in it.
```